### PR TITLE
SL-872

### PR DESCRIPTION
**Pull request by eclipse7**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🔁 CI/CD
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR fixes the "Predictable Choice Bit Leakage in Oblivious Transfer Protocol" vulnerability 
in the Endemic OT implementation.

### Vulnerability Summary:
The vulnerability stems from the unfortunate combination of two factors: 
a misinterpretation of the $\oplus$ operator as bitwise XOR (whereas MR19 uses it to imply curve addition), 
and the choice of abstraction for the elliptic curve underlying the protocol.

The curve used in the sl-crypto implementation was Curve25519, which is
of order 8q where q is a prime. The honest key generation algorithm works
over a specific subgroup of order q, meaning that a randomly chosen point on
the curve is likely to be in the co-domain of honest key generation only with
probability 1/8. Consequently, a passively corrupt sender in the OT protocol
can distinguish the receiver’s honest public key from the random public key with probability 7/8.
This leaks the receiver’s choice bit to the sender with the same probability.

### Applied Fix:
We changed the underlying elliptic curve for the OT from Curve25519 to
secp256k1, which does not have such subtleties with the cofactor; all points
on the elliptic curve are in the co-domain of key generation. Additionally, we
replaced bitwise XOR with subtraction in the curve group.

## Related Tickets
https://slira.atlassian.net/browse/SL-872

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: crates/sl-oblivious/Cargo.toml
Comment (left-arm) [inline crates/sl-oblivious/Cargo.toml:1]: The PR changes API, we have to update version to "0.2.0"

File: crates/sl-oblivious/src/endemic_ot.rs
Comment (left-arm) [inline crates/sl-oblivious/src/endemic_ot.rs:1]: could we use define `type CompressedPoint = [u8; 33];` and use it?

**Review by left-arm — CHANGES_REQUESTED** [review]

**Review by left-arm — APPROVED** [review]
LGTM

**Review by leontiad — APPROVED** [review]
got it

**Pull request merged**